dask/array/tests/test_slicing.py

```diff
@@ -438,6 +438,17 @@ def test_slicing_consistent_names():
     assert same_keys(a[0], a[0])
     assert same_keys(a[:, [1, 2, 3]], a[:, [1, 2, 3]])
     assert same_keys(a[:, 5:2:-1], a[:, 5:2:-1])
+    assert same_keys(a[0, ...], a[0, ...])
+    assert same_keys(a[...], a[...])
+    assert same_keys(a[[1, 3, 5]], a[[1, 3, 5]])
+
+
+def test_slicing_consistent_names_after_normalization():
+    x = da.zeros(10, chunks=(5,))
+    assert same_keys(x[0:], x[:10])
+    assert same_keys(x[0:], x[0:10])
+    assert same_keys(x[0:], x[0:10:1])
+    assert same_keys(x[:], x[0:10:1])
```

Comment:
Should probably check with negative/out-of-bounds slices too:

```python
assert same_keys(x[-11:11], x[:])
assert same_keys(x[-11:-9], x[:1])
```

Reply:
Added

Reply:
I meant to check if the keys were the same (since the oob slices normalize to the boundaries). Not a big deal though.